### ISSUE TITLE
Getting a crash in version 2.7.3 of aws pinpoint

### ISSUE THREAD

**Issue by Pinaki93** [aws-amplify/aws-sdk-android#604]
I'm getting this crash from `UrlHttpClient `, lne 162 on my Crashalytics

```
Fatal Exception: java.lang.ArrayIndexOutOfBoundsException: length=0; index=0
       at com.android.okhttp.internal.Util.rjil_playstore_download_case(Util.java:309)
       at com.android.okhttp.HostResolver$1.getAllByName(HostResolver.java:31)
       at com.android.okhttp.internal.http.RouteSelector.resetNextInetSocketAddress(RouteSelector.java:232)
       at com.android.okhttp.internal.http.RouteSelector.next(RouteSelector.java:124)
       at com.android.okhttp.internal.http.HttpEngine.connect(HttpEngine.java:310)
       at com.android.okhttp.internal.http.HttpEngine.sendRequest(HttpEngine.java:249)
       at com.android.okhttp.internal.http.HttpURLConnectionImpl.execute(HttpURLConnectionImpl.java:397)
       at com.android.okhttp.internal.http.HttpURLConnectionImpl.connect(HttpURLConnectionImpl.java:118)
       at com.android.okhttp.internal.http.HttpURLConnectionImpl.getOutputStream(HttpURLConnectionImpl.java:229)
       at com.android.okhttp.internal.http.DelegatingHttpsURLConnection.getOutputStream(DelegatingHttpsURLConnection.java:218)
       at com.android.okhttp.internal.http.HttpsURLConnectionImpl.getOutputStream(HttpsURLConnectionImpl.java:25)
       at com.amazonaws.http.UrlHttpClient.writeContentToConnection(SourceFile:162)
       at com.amazonaws.http.UrlHttpClient.execute(SourceFile:75)
       at com.amazonaws.http.AmazonHttpClient.executeHelper(SourceFile:371)
       at com.amazonaws.http.AmazonHttpClient.execute(SourceFile:212)
       at com.amazonaws.services.cognitoidentity.AmazonCognitoIdentityClient.invoke(SourceFile:566)
       at com.amazonaws.services.cognitoidentity.AmazonCognitoIdentityClient.getOpenIdToken(SourceFile:510)
       at com.amazonaws.auth.AWSAbstractCognitoIdentityProvider.getToken(SourceFile:198)
       at com.amazonaws.auth.AWSAbstractCognitoIdentityProvider.refresh(SourceFile:316)
       at com.amazonaws.auth.AWSBasicCognitoIdentityProvider.refresh(SourceFile:77)
       at com.amazonaws.mobile.auth.core.IdentityManager$AWSRefreshingCognitoIdentityProvider.refresh(SourceFile:184)
       at com.amazonaws.auth.CognitoCredentialsProvider.startSession(SourceFile:678)
       at com.amazonaws.auth.CognitoCredentialsProvider.getCredentials(SourceFile:465)
       at com.amazonaws.auth.CognitoCachingCredentialsProvider.getCredentials(SourceFile:485)
       at com.amazonaws.auth.CognitoCachingCredentialsProvider.getCredentials(SourceFile:77)
       at com.amazonaws.services.pinpoint.AmazonPinpointClient.invoke(SourceFile:3872)
       at com.amazonaws.services.pinpoint.AmazonPinpointClient.updateEndpoint(SourceFile:3630)
       at com.amazonaws.mobileconnectors.pinpoint.targeting.TargetingClient$1.run(SourceFile:198)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
       at java.lang.Thread.run(Thread.java:818)
```

I'm using Aws Pinpoint version 2.7.3 right now.

**Comment by rohandubal** [comment]
Hello @Pinaki93 

Thanks for reporting this issue. We will look into it.

Thanks,
Rohan

**Comment by rohandubal** [comment]
Hello @Pinaki93 

Could you please update the SDK to latest version? We released a bunch of improvements in `2.8.x`.

Thanks,
Rohan
